homeworks/05-redux-basic/src/reducer.ts

```diff
@@ -0,0 +1,81 @@
+const UPDATE_BALANCE = "UPDATE_BALANCE";
+const CREDIT = "CREDIT";
+const SUBTRACT_PERCENTAGE = "SUBTRACT_PERCENTAGE";
+const DEBIT = "DEBIT";
+
+type State = {
+  balance: number;
+};
+
+const initialState = {
+  balance: 0,
+};
+
+type UpdateBalance = {
+  type: typeof UPDATE_BALANCE;
+  payload: number;
+};
+
+type Credit = {
+  type: typeof CREDIT;
+  payload: number;
+};
+
+type SubtractPercentage = {
+  type: typeof SUBTRACT_PERCENTAGE;
+  payload: number;
+};
+
+type Debit = {
+  type: typeof DEBIT;
+  payload: number;
+};
+
+type Action = UpdateBalance | Credit | SubtractPercentage | Debit;
+
+const updateBalance = (value: number): UpdateBalance => ({
+  type: UPDATE_BALANCE,
+  payload: value,
+});
+
+const credit = (value: number): Credit => ({
+  type: CREDIT,
+  payload: value,
+});
+
+const subtractPercentage = (value: number): SubtractPercentage => ({
+  type: SUBTRACT_PERCENTAGE,
+  payload: value,
+});
+
+const debit = (value: number): Debit => ({
+  type: DEBIT,
+  payload: value,
+});
+
+const reducer = (state = initialState, action: Action): State => {
```

Comment:
Usually actions and reducer are separate files for better readability.
You can split the files or leave it (optional)